airbyte-integrations/connectors/source-github/source_github/streams.py

```diff
@@ -0,0 +1,405 @@
+            params.update(next_page_token)
+
+        return params
+
+    def request_headers(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> Mapping[str, Any]:
+        # Without sending `User-Agent` header we will be getting `403 Client Error: Forbidden for url` error.
+        return {
+            "User-Agent": "PostmanRuntime/7.28.0",
+        }
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        for record in response.json():  # GitHub puts records in an array.
+            yield self.transform(record=record)
+
+    def transform(self, record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Use this method to:
+            - remove excessive fields from record;
+            - minify subelements in the record. For example, if you have `reviews` record which looks like this:
+            {
+              "id": 671782869,
+              "node_id": "MDE3OlB1bGxSZXF1ZXN0UmV2aWV3NjcxNzgyODY5",
+              "user": {
+                "login": "keu",
+                "id": 1619536,
+                ...
+              },
+              "body": "lgtm, just  small comment",
+              "state": "CHANGES_REQUESTED",
+              "html_url": "https://github.com/airbytehq/airbyte/pull/3734#pullrequestreview-671782869",
+              "pull_request_url": "https://api.github.com/repos/airbytehq/airbyte/pulls/3734",
+              "author_association": "MEMBER",
+              "_links": {
+                "html": {
+                  "href": "https://github.com/airbytehq/airbyte/pull/3734#pullrequestreview-671782869"
+                },
+                "pull_request": {
+                  "href": "https://api.github.com/repos/airbytehq/airbyte/pulls/3734"
+                }
+              },
+              "submitted_at": "2021-05-30T01:14:37Z",
+              "commit_id": "be921b0a5a1c5b17539c92420aa231f58c055169"
+            }
+
+            `user` subelement contains almost all possible fields fo user and it's not optimal to store such data in
+            `reviews` record. We may leave only `user.id` field and save in to `user_id` field in the record. So if you
+            need to do something similar with your record you may use this method.
+        """
+        for field in self.object_fields:
+            field_value = record.pop(field, None)
+            record[f"{field}_id"] = field_value.get("id") if field_value else None
+
+        for field in self.list_fields:
+            field_values = record.pop(field, [])
+            record[field] = [value["id"] for value in field_values]
+
+        return record
+
+
+class Assignees(GithubStream):
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return f"repos/{self.repository}/assignees"
```

Comment:
I think it is possible to define a default value for this method in the base class and remove a lot of duplicates.

```suggestion
        return f"repos/{self.repository}/{self.name}"
```

Reply:
You mean `repos/{self.repository}/`?

Reply:
Done.